bansa/src/main/java/com/brianegan/bansa/BansaStore.java

```diff
@@ -0,0 +1,68 @@
+package com.brianegan.bansa;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BansaStore<S, A> implements Store<S, A> {
+    private S currentState;
+    private final Reducer<S, A> reducer;
+    private final List<Subscriber> subscribers = new ArrayList<>();
+    private final List<NextDispatcher<A>> dispatchers = new ArrayList<>();
+    private final Middleware<S, A> dispatcher = new Middleware<S, A>() {
+        @Override
+        public void dispatch(Store<S, A> store, A action, NextDispatcher<A> next) {
+            synchronized (this) {
+                currentState = reducer.reduce(store.getState(), action);
+            }
+            for (int i = 0; i < subscribers.size(); i++) {
+                subscribers.get(i).onStateChange();
```

Comment:
Just thinking if it would be better to receive the current state in onStateChange so we don't need to do getState on the store?